spec/ruby/core/dir/glob_spec.rb

```diff
@@ -122,6 +122,14 @@
     Dir.glob('**/**/**').empty?.should == false
   end
 
+  it "handles simple filename patterns" do
+    Dir.glob('.dotfile').should == ['.dotfile']
+  end
+
+  it "handles simple directory patterns" do
+    Dir.glob('.dotsubdir/').should == ['.dotsubdir/']
```

Comment:
Could you add an example where it doesn't match on a non-directory like `Dir.glob('nondotfile/').should == []`?